packages/react-components/react-utilities/src/hooks/useId.ts

```diff
@@ -19,11 +19,20 @@ export function resetIdsForTests(): void {
 export function useId(prefix: string = 'fui-', providedId?: string): string {
   const contextValue = useSSRContext();
 
-  return React.useMemo(() => {
-    if (providedId) {
-      return providedId;
-    }
+  // Checking if useId is available on React, if it is, we use it to generate the id. String concatenation is used to
+  // prevent bundlers from complaining with older versions of React.
+  const _useId: () => string = (React as never)['use' + 'Id'];
```

Comment:
```suggestion
  const _useId: (() => string) | undefined = (React as never)['use' + 'Id'];
```

nit: Let's make types more accurate